mods/default/functions.lua

```diff
@@ -757,3 +757,34 @@ function default.can_interact_with_node(player, pos)
 
 	return false
 end
+
+local non_player_action_log = minetest.settings:get_bool("enable_non_player_action_log") or false
+
+function default.log_action(player, pos, message)
+	local who = player and player:get_player_name() or "(something)"
```

Comment:
Please remove the `or "(something)"` as it wasn't there before either; a valid `player` object must have a name (or else it *should* throw the "attempt to concat a nil val" err). I'm also not sure whether `nil` players should be allowed here.